rest_framework/serializers.py

```diff
@@ -973,12 +973,17 @@ def update(self, instance, validated_data):
         # Note that unlike `.create()` we don't need to treat many-to-many
         # relationships as being a special case. During updates we already
         # have an instance pk for the relationships to be associated with.
+        m2m_fields = []
         for attr, value in validated_data.items():
-            if attr in info.relations and info.relations[attr].to_many:
-                field = getattr(instance, attr)
-                field.set(value)
-            else:
+            if attr not in info.relations or not info.relations[attr].to_many:
```

Comment:
I would revert the condition, as the previous version is more readable to me.

Reply:
Good point, I've reverted the condition in the 064f1e9a146e323f6c6e7c252243ac4702827a38.